pharmaverse/aNCA
# Enhancement: better slope selector plots

## Issue

Closes #23, #22 and #36 

## Description

- Split slope selection functionality into separate module
- Improved UI and UX for the slope exclusion functionality
- Fixed help widget for the module
- Introduces `shinyjs`, `reactable` and `reactable.extras` packages to the app
- Refactored code, moved logic into separate functions, streamlined data flow

## Definition of Done

- [x] slope selector functionality migrated to separate module
- [x] addressed complexity of server function for the module
- [x] fixed warnings generated by `ggplot2` and `plotly`
- [x] fixed overlapping annotations on the plots
- [x] fixed/refactored help widget
- [x] improved page layout for plots
- [x] updated gifs in manual widget

## How to test

Open the application and run NCA as normal. Go to Setup -> Slope Selector tab and play around with the plots. You can provide selections and exclusions either by data table or by clicking on the plots themselves. After each change, the plots should update to reflect it. To run the analysis again and apply ruleset to the actual data, the **Apply** button will need to be clicked.

## Contributor checklist
